examples/acados_python/quadrotor_nav/acados_settings.py

```diff
@@ -0,0 +1,229 @@
+
+        ineq_constr_eqn = []
+        ineq_constr_eqn = ca.vertcat(ineq_constr_eqn, dyn_constr_eqn)
+
+        model_ac.con_h_expr = ineq_constr_eqn
+        model_ac.con_h_expr_e = ineq_constr_eqn
+
+        # Inequality bounds
+        nh = model_ac.con_h_expr.shape[0]
+
+        # # Constrain controls
+        # lbu = [0] * self.nu;      ubu = [0] * self.nu
+
+        # # Control bounds ( Affects horizon quality before switch)
+        # lbu[0] = OHM_MIN;         ubu[0] = OHM_MAX
+        # lbu[1] = OHM_MIN;         ubu[1] = OHM_MAX
+
+        # ocp.constraints.lbu = np.array(lbu)
+        # ocp.constraints.ubu = np.array(ubu)
+        # ocp.constraints.idxbu = np.array([0, 1])
+
+        # Bounds on path constraints (inequality)
+        lh = np.zeros(nh);        uh = np.zeros(nh)
+        lh[:] = -INF;             uh[:] = 1
+
+        ocp.constraints.lh = lh
+        ocp.constraints.uh = uh
+
+        ocp.constraints.lh_e = lh
+        ocp.constraints.uh_e = uh
+
+        # Configure itegrator and QP solver
+        ocp.solver_options.integrator_type = "ERK"
+        ocp.solver_options.tf = Tf
+        ocp.solver_options.sim_method_num_stages = 4
+        ocp.solver_options.sim_method_num_steps = 1
+        # ocp.solver_options.collocation_type = 'GAUSS_RADAU_IIA'
+        # ocp.solver_options.time_steps = time_steps
+        # ocp.solver_options.shooting_nodes = shooting_nodes
+
+        ocp.solver_options.qp_solver = "PARTIAL_CONDENSING_HPIPM"#"PARTIAL_CONDENSING_HPIPM" #"FULL_CONDENSING_HPIPM" #"PARTIAL_CONDENSING_HPIPM"
+        ocp.solver_options.hessian_approx =  "GAUSS_NEWTON"#"EXACT",
+        # ocp.solver_options.cost_discretization ="INTEGRATOR"
+        ocp.solver_options.nlp_solver_type = "SQP_RTI"
+        ocp.solver_options.tol = 1e-3
+        ocp.qp_solver_tol = 1e-3
+
+        # create solver
+        solve_json = "planner_ocp.json"
+        self.ocp = ocp
+        self.solver = AcadosOcpSolver(ocp, json_file = solve_json)
+        self.integrator = AcadosSimSolver(ocp, json_file = solve_json) #TODO
+
+        return True
+
+
+    def solve_and_sim(self):
+        '''Solve the OCP with multiple shooting, and forward simulate with RK4'''
+
+        # Integrate ODE model to get CL estimate (no measurement noise)
+        u_0, status = self.solve_for_x0(x0_bar = self.zeta_0)
+        self.zeta_0 = self.integrator.simulate(x=self.zeta_0, u=u_0)
+
+        self.zeta_N = np.reshape(self.solver.get(0, "x"), (self.nx, 1))
+        for i in range(1, N +1):
+            zeta_i = np.reshape(self.solver.get(i, "x"), (self.nx, 1))
+            self.zeta_N = np.concatenate((self.zeta_N, zeta_i), axis = 1)
+
+        self.u_N[:, 0] = u_0
+        return status
+
+    # user expected to handle status, instead of raising exception
+    def solve_for_x0(self, x0_bar):
+        '''Wrapper around `solve()` which sets initial state constraint, solves the OCP, and returns u0.'''
+
+        self.solver.set(0, "lbx", x0_bar)
```

Comment:
Maybe use `solve_for_x0` provided by the acados solver here?
https://github.com/acados/acados/blob/90607d14d7b7dd9b71d6e201a7df99e5eda16815/interfaces/acados_template/acados_template/acados_ocp_solver.py#L368